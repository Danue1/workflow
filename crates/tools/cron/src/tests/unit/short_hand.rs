macro_rules! snapshot {
    ($( $name:ident -> $source:expr, )+) => {
        $(
            #[test]
            fn $name() {
                let result = crate::Schedule::parse($source);
                insta::assert_debug_snapshot!(result);
            }
        )+
    };
}

snapshot! {
    secondly -> "@secondly",
    minutely -> "@minutely",
    hourly -> "@hourly",
    daily -> "@daily",
    weekly -> "@weekly",
    sunday -> "@sunday",
    monday -> "@monday",
    tuesday -> "@tuesday",
    wednesday -> "@wednesday",
    thursday -> "@thursday",
    friday -> "@friday",
    saturday -> "@saturday",
    monthly -> "@monthly",
    even_monthly -> "@even-monthly",
    odd_monthly -> "@odd-monthly",
    yearly -> "@yearly",
    annually -> "@annually",
    even_yearly -> "@even-yearly",
    odd_yearly -> "@odd-yearly",
    leap_yearly -> "@leap-yearly",
}
