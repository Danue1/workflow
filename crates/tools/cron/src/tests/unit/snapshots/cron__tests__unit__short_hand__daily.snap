---
source: crates/tools/cron/src/tests/unit/short_hand.rs
expression: result
---
Ok(
    (
        "",
        Schedule {
            source: "@daily",
            second: SecondRule {
                kinds: [
                    SecondKind {
                        range: (
                            Second(
                                0,
                            ),
                            Second(
                                0,
                            ),
                        ),
                        periodic: SecondPeriodic(
                            1,
                        ),
                    },
                ],
            },
            minute: MinuteRule {
                kinds: [
                    MinuteKind {
                        range: (
                            Minute(
                                0,
                            ),
                            Minute(
                                0,
                            ),
                        ),
                        periodic: MinutePeriodic(
                            1,
                        ),
                    },
                ],
            },
            hour: HourRule {
                kinds: [
                    HourKind {
                        range: (
                            Hour(
                                0,
                            ),
                            Hour(
                                0,
                            ),
                        ),
                        periodic: HourPeriodic(
                            1,
                        ),
                    },
                ],
            },
            date: DateRule {
                kinds: [
                    DateKind {
                        range: (
                            Date(
                                1,
                            ),
                            Date(
                                31,
                            ),
                        ),
                        periodic: DatePeriodic(
                            1,
                        ),
                    },
                ],
            },
            week: WeekRule {
                kinds: [
                    WeekKind {
                        range: (
                            Sunday,
                            Saturday,
                        ),
                        periodic: WeekPeriodic(
                            1,
                        ),
                    },
                ],
            },
            month: MonthRule {
                kinds: [
                    MonthKind {
                        range: (
                            January,
                            December,
                        ),
                        periodic: MonthPeriodic(
                            1,
                        ),
                    },
                ],
            },
            year: YearRule {
                kinds: [
                    YearKind {
                        range: (
                            Year(
                                1970,
                            ),
                            Year(
                                2199,
                            ),
                        ),
                        periodic: YearPeriodic(
                            1,
                        ),
                    },
                ],
            },
        },
    ),
)
